packages/sync/src/Replicastore.php

```diff
@@ -1202,6 +1202,38 @@ public function get_checksum_columns_for_object_type( $object_type ) {
 		}
 	}
 
+	/**
+	 * Grabs the minimum and maximum object ids for the given parameters.
+	 *
+	 * @access public
+	 *
+	 * @param int    $id_field     The id column in the table to query.
+	 * @param bool   $distinct     If true, will query for a distinct object id.
+	 * @param string $object_table The table to query.
+	 * @param string $where        A sql where clause without 'WHERE'.
+	 * @param int    $bucket_size  The maximum amount of objects to include in the query.
+	 *
+	 * @return object An object with min_id and max_id properties.
+	 */
+	public function get_min_max_object_id( $id_field, $distinct, $object_table, $where, $bucket_size ) {
```

Comment:
There are a couple of problems I see in this method:

* It's a bit complex to understand what it's doing because of the complex logic in between the SQL partials.
* We're not properly escaping the SQL in some places, and that's not entirely visible by the way the code is written.

Could we separate this in a couple of methods that could help fix both of the above problems? I'm fine with some code repetition in favor of more secure and readable code, if that makes sense.

Reply:
It does make sense, but i cannot figure out how to do it :(

Reply:
it is a hard one, there's a lot that we don't prepare from previous code, @tyxla would you like to give it a try?

Reply:
I agree there's a lot we don't prepare in previous code, but I was hoping we'll resolve these cases instead of introducing new ones. Unprepared/unescaped code is prone to injections, and I am really hesitant to ship any new vulnerabilities. 

I'm happy to take over and give that a try; but to achieve it, I'd also have to refactor some other things, as suggested in my other reviews (move some code, change some methods and implementations). Would you be fine with that @roccotripaldi @lezama? I appreciate the hard work done here and didn't mean to throw any of it away by refactoring a big piece of it.

Reply:
Taken a quick look about how we can `prepare` this one, and it's not quite easy right now. We'd have to refactor a bunch of other things in sync, and it feels like this will have to be a separate effort. So I've added a `@todo` for now, and I suggest we circle back to it at some point.